File: search-server/src/main/java/com/example/search/domain/Product.java
package com.example.search.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

@Document(indexName = "product")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    private String id;
    private String name;
    private Double price;
    private String category;
}

File: search-server/src/main/kotlin/com/example/search/domain/Product.kt
package com.example.search.domain

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document

@Document(indexName = "product")
class Product(
    @Id
    var id: String? = null,
    var name: String? = null,
    var price: Double? = null,
    var category: String? = null
)
